### PR TITLE
Validate Binary Search Tree in C++, C, Java, and Python.

### LINKED ISSUE #1131: [SSoC'23] Validate binary Search Tree
First, we define the TreeNode structure, which represents a node in the BST. It has three members: val to store the node's value, and left and right pointers to the left and right child nodes, respectively.

The isValidBSTHelper function is a recursive helper function that takes a TreeNode* as input along with the minimum and maximum values that the node's value should fall between. It performs the following checks:

If the node is nullptr, it means we have reached the end of a subtree, so we return true (since an empty subtree is considered a valid BST).
If the node's value is less than or equal to the minimum value or greater than or equal to the maximum value, it violates the BST property, so we return false.
We recursively call isValidBSTHelper for the left and right subtrees, updating the minimum and maximum values accordingly. If both subtrees return true, the current subtree is a valid BST, so we return true.
The isValidBST function is the main function that calls the isValidBSTHelper function with the root node of the BST and the minimum and maximum values (LLONG_MIN and LLONG_MAX) as initial bounds.

The insert function is used to dynamically construct the BST based on the user's input. It takes the root node and the value to be inserted as parameters. If the root node is nullptr, it means the tree is empty, so a new node is created with the given value and returned. Otherwise, based on the value being less than or greater than the root node's value, the function is recursively called on the left or right subtree, respectively. The function then updates the left or right child pointer of the root node accordingly.

The deleteTree function is a recursive function that deallocates the memory allocated for the BST nodes. It takes the root node as input and performs a post-order traversal to delete the nodes. First, it recursively calls deleteTree for the left and right subtrees, and then it deletes the current node.

In the main function, we start by declaring the root node as nullptr and a variable value to store the user's input. We prompt the user to enter values to construct the BST, and the input loop continues until the user enters -1. Inside the loop, we call the insert function to insert the entered value into the BST.

After constructing the BST, we call the isValidBST function to check if the BST is valid. If it is, we print a message indicating that the BST is valid; otherwise, we print a message indicating that it's not valid.

Finally, we clean up the memory allocated for the BST by calling the deleteTree function, passing the root node as the parameter.

### PR DESCRIPTION
**Pull request by mohitsingh1011**
Closes #1131 

## 🛠️ Issue (1131)

- [✔️] My code follows the code style of this project.
- [✔️ ] This PR does not contain plagiarized content.
- [✔️ ] The title of my pull request is a short description of the requested changes.

## 📷 Screenshots

C++
[
image
](url)

C

image

Java

image

Python

image

### CODE REVIEW

**Review by Kumar-laxmi — CHANGES_REQUESTED** [review]
Place the Java's program file inside Tree folder

**Comment by mohitsingh1011** [comment]
@Kumar-laxmi changes made please review it.

**Review by zugzwang03 — APPROVED** [review]

**Review by 072arushi — APPROVED** [review]

**Review by Arun9739 — APPROVED** [review]

**Comment by MohitSingh101** [comment]
@Kumar-laxmi please merge pr.

**Comment by mohitsingh1011** [comment]
@Kumar-laxmi @Arun9739 @072arushi @zugzwang03 please add the Ready to Merge label.

**Comment by mohitsingh1011** [comment]
@Kumar-laxmi please review and merge.

**Review by Kumar-laxmi — APPROVED** [review]

File: Java/Trees/Validate_binary_Search_Tree.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
+import java.io.*;
+ 
+class GFG {
```
Comment (Kumar-laxmi) [inline Java/Trees/Validate_binary_Search_Tree.java:3]: This clearly shows the code has been plagarised from: https://www.geeksforgeeks.org/a-program-to-check-if-a-binary-tree-is-bst-or-not/

**Comment by Kumar-laxmi** [comment]
Do you have any explanation for this

File: Java/Trees/Validate_binary_Search_Tree.java (hunk 1 above)
Comment (MohitSingh101) [reply Java/Trees/Validate_binary_Search_Tree.java:3]: Yes I have referred GFG if needed class name can be changed.

Comment (Kumar-laxmi) [reply Java/Trees/Validate_binary_Search_Tree.java:3]: @mohitsingh1011 I think you are not aware that plagarism in any Open-source event can lead to deduction of points and repeated violation can lead to disqualification.

Comment (Kumar-laxmi) [reply Java/Trees/Validate_binary_Search_Tree.java:3]: There is a difference between referring and direct copy-pasting

**Pull request closed without merging**